src/core/createCookie.js

```diff
@@ -22,55 +22,47 @@ const createCookie = (prefix, id, cookieDomain = "") => {
      */
     get(name) {
       const cookieName = `${ALLOY_COOKIE_NAME}_${id}`;
-      let currentCookie;
-      try {
-        currentCookie = cookie.getJSON(cookieName);
-        return currentCookie[prefix][name];
-      } catch (error) {
-        return currentCookie;
-      }
+      const currentCookie = cookie.get(cookieName);
+      const currentCookieParsed = currentCookie && JSON.parse(currentCookie);
+      return (
+        currentCookieParsed &&
+        currentCookie[prefix] &&
+        currentCookie[prefix][name]
+      );
     },
     /**
      * Updates and returns values from AlloyCookie for Components
      * @param {...*} arg Strings with key and value to be stored in alloy cookie.
      */
     set(key, value) {
       const cookieName = `${ALLOY_COOKIE_NAME}_${id}`;
-      let currentCookie;
-      try {
-        currentCookie = JSON.parse(cookie.get(cookieName));
-      } catch (error) {
-        console.log("catch");
-        currentCookie = {};
-      }
-      console.log(2, JSON.stringify(currentCookie));
+      const currentCookie = cookie.get(cookieName)
+        ? JSON.parse(cookie.get(cookieName))
```

Comment:
Same here.

Reply:
here also if `cookie.get(cookieName)` is `undefined` it will not do `JSON.parse`